dot/utils_integration_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+//go:build integration
+// +build integration
+
+package dot
+
+import (
+	"log"
+	"os"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/stretchr/testify/require"
+)
+
+// TestNewConfig tests the NewTestConfig method
+func TestNewConfig(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+}
+
+// TestNewConfigAndFile tests the NewTestConfigWithFile method
+func TestNewConfigAndFile(t *testing.T) {
+	testCfg, testCfgFile := newTestConfigWithFile(t)
+	require.NotNil(t, testCfg)
+	require.NotNil(t, testCfgFile)
+}
+
+// TestInitNode
+func TestNewTestGenesis(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genFile := NewTestGenesisRawFile(t, cfg)
+	require.NotNil(t, genFile)
+
+	cfg.Init.Genesis = genFile
+}
+
+func TestNewTestGenesisFile(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
+
+	genHRFile := newTestGenesisFile(t, cfg)
+	require.NotNil(t, genHRFile)
+	defer os.Remove(genHRFile)
+
+	genRawFile := NewTestGenesisRawFile(t, cfg)
+	require.NotNil(t, genRawFile)
+	defer os.Remove(genRawFile)
+
+	genHR, err := genesis.NewGenesisFromJSON(genHRFile, 0)
+	require.NoError(t, err)
+	genRaw, err := genesis.NewGenesisFromJSONRaw(genRawFile)
+	require.NoError(t, err)
+
+	// values from raw genesis file should equal values generated from human readable genesis file
+	require.Equal(t, genRaw.Genesis.Raw["top"], genHR.Genesis.Raw["top"])
+}
+
+func TestDeepCopyVsSnapshot(t *testing.T) {
```

Comment:
Please remove that test, it got removed elsewhere in a PR a few weeks ago.

Reply:
Removed.